tools/generator/filters/caml.py

```diff
@@ -0,0 +1,87 @@
+import textwrap
+
+from . import register_filter
+from .common import is_array
+
+
+@register_filter
+def caml_type(value: str) -> str:
+    if value == 'void':
+        return 'unit'
+    if value == 'double':
+        return 'float'
+    if is_array(value):
+        return '{} array'.format(value)
+    return value
+
+
+@register_filter
+def caml_callback(func) -> str:
+    '''symbolic name of a function to be used with Callback.register in caml
+    code, and with caml_named_value in C code
+    '''
+    return 'ml_' + func['fct_name']
+
+
+# caml functions need to take at least one parameter, be it a unit value if
+# there is nothing useful to give
+
+@register_filter
+def cxx_nonempty_or_unit(arg_list):
+    if not arg_list:
+        return [('unit', 'void', 'empty caml parameter')]
+    return arg_list
+
+
+@register_filter
+def caml_prototype(func) -> str:
+    arg_list = ' '.join(arg_name for arg_name, _, _ in func['fct_arg'])
+    if arg_list == '':  # make unit the only parameter
```

Comment:
Can't you just pass your arglist through `nonempty_or_unit` first to avoid specialcasing?

Reply:
I would need to make an other filter to handle empty parameter list in ocaml functions.